include/hdf5_particle.h

```diff
@@ -39,8 +38,33 @@ typedef struct HDF5Particle {
   mpm::Index cell_id;
   // Status
   bool status;
+  // Material id
+  unsigned material_id;
+  // Number of state variables
+  unsigned nstate_vars;
+  // State variables (init to zero)
+  double svars[20] = {0};
 } HDF5Particle;
 
+namespace hdf5::particle {
+const hsize_t NFIELDS = 53;
```

Comment:
Also, this NFIELDS. Are you fixing the total number of fields printed? This means that for 2 or 3 phase particles, with more variables, we need to derive this `HDF5Particle` struct right? Isn't it easier and better to make the struct to be implemented inside the `particle` class then?

Reply:
We can't have the `struct` defined inside the `Particle` because we are serializing the `Particle` class. Yes, we need to derive `HDF5Particle` for 2 and 3 phase

Reply:
Okay. I am sorry. You are right. I realized that it's also harder to access from `mesh` if we do that.